pkg/objstore/s3/s3.go

```diff
@@ -271,21 +286,42 @@ func NewTestBucket(t testing.TB, location string) (objstore.Bucket, func(), erro
 		return nil, nil, err
 	}
 
+	if c.Bucket != "" {
+		if os.Getenv("THANOS_ALLOW_EXISTING_BUCKET_USE") == "" {
+			return nil, nil, errors.New("S3_BUCKET is defined. Normally this tests will create temporary bucket " +
+				"and delete it after test. Unset S3_BUCKET env variable to use default logic. If you really want to run " +
+				"tests against provided (NOT USED!) bucket, set THANOS_ALLOW_EXISTING_BUCKET_USE=true. WARNING: That bucket " +
+				"needs to be manually cleared. This means that it is only useful to run one test in a time. This is due " +
+				"to safety (accidentally pointing prod bucket for test) as well as aws s3 not being fully strong consistent.")
+		}
+
+		if err := b.Iter(context.Background(), "", func(f string) error {
+			return errors.Errorf("bucket %s is not empty bucket", c.Bucket)
+		}); err != nil {
+			return nil, nil, errors.Wrap(err, "s3 check bucket")
```

Comment:
I meant "s3 check bucket" should be "failed checking s3 bucket %s"

Reply:
it should not. I already said why.